pkg/cache/cluster.go

```diff
@@ -661,13 +661,34 @@ func (c *clusterCache) IsNamespaced(gk schema.GroupKind) (bool, error) {
 	return false, errors.NewNotFound(schema.GroupResource{Group: gk.Group}, "")
 }
 
+func (c *clusterCache) managesNamespace(namespace string) bool {
+	if len(c.namespaces) == 0 {
+		return true
+	}
+	if len(namespace) == 0 {
+		return true
```

Comment:
I think we should return false here. If `len(namespace) == 0` is true then resource is a cluster-level resource. Cluster level resources are not managed if cluster is connected in namespaced mode so we should return false.

```golang
if namespace == "" {
	return true
}
```

Reply:
I've made the suggested change.